txpool/txpool_test.go

```diff
@@ -116,6 +117,55 @@ func TestMultipleTransactions(t *testing.T) {
 	assert.Equal(t, pool.Length(), uint64(1))
 }
 
+func TestGetQueuedAndPendingTransactions(t *testing.T) {
+	pool, err := NewTxPool(hclog.NewNullLogger(), false, forks.At(0), &mockStore{}, nil, nil)
+	assert.NoError(t, err)
+	pool.EnableDev()
+
+	from1 := types.Address{0x1}
+	txn0 := &types.Transaction{
+		From:     from1,
+		Nonce:    0,
+		Gas:      validGasLimit,
+		Value:    big.NewInt(106),
+		GasPrice: big.NewInt(1),
+	}
+	assert.NoError(t, pool.addImpl("", txn0))
+
+	from2 := types.Address{0x2}
+	txn1 := &types.Transaction{
+		From:     from2,
+		Nonce:    1,
+		Gas:      validGasLimit,
+		Value:    big.NewInt(106),
+		GasPrice: big.NewInt(1),
+	}
+	assert.NoError(t, pool.addImpl("", txn1))
+
+	txn2 := &types.Transaction{
+		From:     from2,
+		Nonce:    2,
+		Gas:      validGasLimit,
+		GasPrice: big.NewInt(1),
+	}
+	assert.NoError(t, pool.addImpl("", txn2))
+
+	from3 := types.Address{0x3}
+	txn3 := &types.Transaction{
+		From:     from3,
+		Nonce:    3,
+		Gas:      validGasLimit,
+		GasPrice: big.NewInt(1),
+	}
+	assert.NoError(t, pool.addImpl("", txn3))
+
+	queuedTxs, pendingTxs := pool.GetTxs()
+
+	assert.Len(t, queuedTxs, 2)
```

Comment:
Shouldn't there be 3 queued transactions (`txn1, txn2 and txn3`)?

Reply:
Yes, however in this test 2 txs were sent from 1 account. I now separated the `from` of each tx so that the length of the queued map is 3 which makes it easier to follow.